src/bridge/cockpitdbusjson.c

```diff
@@ -2933,7 +2950,43 @@ cockpit_dbus_json_prepare (CockpitChannel *channel)
         }
       else
         {
-          g_bus_get (self->bus_type, self->cancellable, on_bus_ready, g_object_ref (self));
+          const gchar *key;
+
+          if (self->bus_type == G_BUS_TYPE_SYSTEM)
+            key = "system-bus";
+          else if (self->bus_type == G_BUS_TYPE_SESSION)
+            key = "session-bus";
+          else
+            g_assert_not_reached ();
```

Comment:
This is duplicated; I suggest moving into a static helper function, just in case we need to support arbitary other buses in the future.